### PR TITLE
Update documentation for Folder and Organization IAM resources.

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/6093 
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/6220

I wasn't able to autogen the docs so I wrote it manually.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
resource-manager: updated documentation for `folder_iam_*` and `organization_iam_*` resources.
```

### CODE REVIEW

**Comment by ndmckinley** [comment]
Hi,

Sorry, this diff is huge.  Can you explain in more detail what you are trying to do?

**Comment by upodroid** [comment]
Hi, the IAM docs for Folder and Organizations are in the old style instead of a single page like you have for projects and subnets. I wasn't able to auto generate it so I manually write the docs and deleted the old pages.

**Comment by ndmckinley** [comment]
Oh, okay, that seems great!  Let me just make sure this works the way I think it probably does.  :)

File: products/resourcemanager/terraform.yaml [hunk 1]
```diff
@@ -37,6 +37,17 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       decoder: templates/terraform/decoders/avoid_meaningless_project_update.erb
   Folder: !ruby/object:Overrides::Terraform::ResourceOverride
     exclude: true
+    # iam_policy: !ruby/object:Api::Resource::IamPolicy
```
Comment (ndmckinley) [inline products/resourcemanager/terraform.yaml:40]: Hm, I don't see what commenting this out does to the downstream diffs - wondering, can you revert this change?

Comment (upodroid) [reply products/resourcemanager/terraform.yaml]: I got the autogen working now, but it has incorrect logic. Folders and Organizations IAM can't use custom project roles but the doc has that in there.

**Comment by ndmckinley** [comment]
Hi,

Hm, I think I might have lost track of your goals with this PR.  Would you mind filing an issue so I can see what you're trying to accomplish specifically?  The docs change you were making at first was great, if you wanted to go back to just making that change that would be very helpful, or, I'm open to a more expansive change if you have an idea.

I think that the reason that the resources are not currently autogenerated is they don't support everything that a normal IAM resource would, like custom roles.  A few notes if you do want to go down that path:
* if you are generating those resources, you'll need to delete the handwritten ones from `third_party/terraform`
* the docs can be modified using the templates in `templates/terraform` - but be careful, because they're complicated and big changes can have side effects on other resources.

**Comment by upodroid** [comment]
The linked issue needs the docs to show that conditions can be applied to folder iam. I'll achieve this by writing the document manually like we do for project iam.

**Comment by ndmckinley** [comment]
Aha, I see!  Okay, that's great.

**Comment by upodroid** [comment]
We can merge it now.

**Comment by ndmckinley** [comment]
I'm not sure - it looks like this diff removes the docs for the folder resource, which I don't think is intentional.

**Comment by upodroid** [comment]
My bad, restored the deleted file

**Comment by ndmckinley** [comment]
Hm, organization's individual docs are deleted as well - did you mean to handwrite that and it's missing here, or would you like to revert the deletions?

**Comment by upodroid** [comment]
I wrote that too, it got deleted midway. Adding it back now
